projects/ui-components/lf-login/lf-login.component.ts

```diff
@@ -186,13 +188,14 @@ export class LfLoginComponent implements OnChanges, OnDestroy {
         return undefined;
       }
       else {
-        const response = await this.refreshAsync();
-        const newAccessToken = await this.loginService.parseTokenResponseAsync(response, true);
-        if (newAccessToken) {
+        try {
+          const response = await this.loginService.tokenClient?.refreshAccessToken(refreshToken, this.client_id);
+          const newAccessToken = await this.loginService.parseTokenResponseAsync(response!);
```

Comment:
do we know token client will exist, maybe make that ! instead of response (if it will let you)

Reply:
To make sure tokenClient always exist,what do you think of assigning an initial value with the default authorize_url_host_name (laserfiche.com)?

Reply:
response will exist if the call returns 200. If there is error, refreshAccessToken will throw error (but not always HTTPError). yeah the error handling should account for the case when it doesn't throw HTTPError

Reply:
yeah I think before it would have  been handled in the higher catch so maybe just throw it?
so that it can log out, etc.